### PR TITLE
sped up the reseeding process

### PR DESCRIPTION
**Pull request by SandyYuan**
The previous reseeding implementation significantly degrades performance. The new implementation uses njit parallel for significant speedups.

### CODE REVIEW

**Comment by SandyYuan** [comment]
Now using the new parallel RNG.

File: abacusnbody/hod/abacus_hod.py [hunk 1]
```diff
@@ -522,9 +523,28 @@ def staging(self):
...
+    @njit(parallel = True, fastmath = True)
+    def _np_randn_parallel(N, Nthread, reseed):
+        numba.set_num_threads(Nthread)
+        np.random.seed(reseed)
+        return np.random.randn(N)
 
+
+    # np_random_parallel = numba.njit(parallel=True)(self, lambda N: np.random.random(N))
+
+    # np_randn_parallel = numba.njit(parallel=True)(self, lambda N: np.random.randn(N))
```
Comment (lgarrison) [inline abacusnbody/hod/abacus_hod.py:544]: This can all go away now, right?

**Review by lgarrison — CHANGES_REQUESTED** [review]
Just one minor question about leftover code. Besides that, it looks like there might be some empty h5 files in the test dir?

File: abacusnbody/hod/abacus_hod.py (hunk 1 above)
Comment (SandyYuan) [reply abacusnbody/hod/abacus_hod.py]: What empty h5 files?

Comment (lgarrison) [reply abacusnbody/hod/abacus_hod.py]: Oops! I read the GitHub diff "+0 bytes" as meaning "0 bytes". Looks good.

**Comment by lgarrison** [comment]
Summarizing offline discussions, we resolved this performance issue with https://github.com/lgarrison/parallel-numpy-rng.

**Comment by lgarrison** [comment]
Oops, forgot a changelog entry. I'll add one in master.

**Pull request merged**